src/solver.rs

```diff
@@ -199,6 +200,46 @@ pub enum Dependencies<P: Package, VS: VersionSet, M: Eq + Clone + Debug + Displa
     Available(DependencyConstraints<P, VS>),
 }
 
+/// Some statistics about how much trouble the resolver has had with a package.
+pub struct PackageResolutionStatistics {
+    discovery_order: u32,
+    conflict_count: u32,
+}
+
+impl PackageResolutionStatistics {
+    fn new<P: Package>(pid: Id<P>, conflict_count: &Map<Id<P>, u32>) -> Self {
+        Self {
+            discovery_order: pid.into_raw() as u32,
+            conflict_count: conflict_count.get(&pid).cloned().unwrap_or_default(),
+        }
+    }
+
+    /// The number of packages known by Pubgrub when this package was mentioned for the first time.
+    ///
+    /// The root package will return `0`. It's direct dependencies will start at `1` and go up from there.
+    /// Prioritizing based on this value directly will lead to a depth first search of the resolution graph.
+    /// Prioritizing based on the reverse of this value will lead to a breadth first search of the resolution graph.
+    ///
+    /// Note: The exact values depend on implementation details of PubGrub and its dependencies.
+    /// So should not be relied on and may change between any lock file update.
+    pub fn discovery_order(&self) -> u32 {
+        self.discovery_order
+    }
+
+    /// The number of times this package was involved in a conflict that caused a back jump.
+    ///
+    /// When resolution is proceeding normally, this value will stay at `0` for all packages.
+    /// Therefore, using this for prioritization will not affect the properties of simple cases
+    /// like checking a lock file.
+    /// Prioritizing based on this value directly allows the resolver to focus on the packages
+    /// it is having the most problems with.
+    ///
+    /// Note: The exact values depend on implementation details of PubGrub. So should not be relied on and may change.
```

Comment:
We should be clearer about what this value means. Something like:

> Every time an incompatibility is fulfilled, the package involved are marked as conflicting. PubGrub builds incompatibilities lazily, and there construction may change over time, so the value may also change across PubGrub releases.

Reply:
I don't know that a user would know what a `incompatibility` is. I'm not finding a good balance between too much detail and explaining why things could change.

Reply:
I think a consumer of the library probably needs to know what an incompatibility is, right?

Reply:
Not every user. But one that start playing with conflict counts should definitely IMO.

Reply:
We can use a two-tiered explanantion: For basic users, this number is a measure for conflictingn-ess, the higher it is, the more you want to prioritize the package. For advanced users, it's helpful to know that this is the number of incompatibilities that were satisfied that this package was involved in, so a user can implement heuristics on top of it (as we want in uv), e.g. weighing of bounds-strictness vs. conflicting-ness.